src/Ssh/Ssh/Common/SshBaseCmdlet.cs

```diff
@@ -432,24 +452,47 @@ protected internal void SetResourceType()
             WriteProgress(record);
         }
 
-        protected internal void GetRelayInformation()
+        protected internal EndpointAccessResource GetRelayInformation()
         {
-            string _exception = "";
-            if (ResourceId != null)
+            SetResourceId();
+            EndpointAccessResource cred = null;
+
+            try
             {
-                relayInformationResource = RelayInformationUtils.GetRelayInformation(ResourceId, out _exception);
+                ListCredentialsRequest req = new ListCredentialsRequest(serviceName: "SSH");
+                cred = EndpointsClient.ListCredentialsWithHttpMessagesAsync(ResourceId, "default", 3600, req)
+                    .GetAwaiter().GetResult().Body;
             }
-            else
+            catch (PowerShell.Ssh.Helpers.HybridConnectivity.Models.ErrorResponseException exception)
             {
-                relayInformationResource = RelayInformationUtils.GetRelayInformation(ResourceGroupName, Name, ResourceType, out _exception);
-            }
+                // These operations will not succeed if the user isn't an Owner/Contributor.
+                if (exception.Response.StatusCode == HttpStatusCode.PreconditionFailed)
+                {
+                    // Service Configuration missing
+                    // attempt to create it if the user has permission
+                    CreateServiceConfiguration();
+                }
+                else if (exception.Body.Error.Code.Equals("ResourceNotFound"))
```

Comment:
Do we know that these exception properties are always non-null?  Should we use the pattern:
```csharp
else if (exception.Body?.Error?.Code?.Equals("ResourceNotFound"))
```

Also should the string compare be case insensitive?